database/src/main/scala/cromwell/database/slick/JobStoreSlickDatabase.scala

```diff
@@ -10,23 +10,36 @@ trait JobStoreSlickDatabase extends JobStoreSqlDatabase {
 
   import dataAccess.driver.api._
 
-  override def addJobStoreEntries(entries: Iterable[JobStoreEntry])(implicit ec: ExecutionContext): Future[Unit] = {
-    val action = dataAccess.jobStoreAutoInc ++= entries
+  private def addJobStoreEntry(jobStoreEntry: JobStoreEntry, jobStoreSimpletonEntries: Iterable[JobStoreResultSimpletonEntry])
+                              (implicit ec: ExecutionContext): DBIO[Unit] = {
+    for {
+      insertJobStoreEntry <- dataAccess.jobStoreAutoInc += jobStoreEntry
+      entries = jobStoreSimpletonEntries map { _.copy(jobStoreId = insertJobStoreEntry.self) }
```

Comment:
@kshakir does this look right to you?  This Slick `SingleInsertResult` stuff is all a bit magical to me.  😦